[HOLD] Handle errors on manual requests

neil-marcellini opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
Hello reviewer, thanks for taking a look. Here we are handling errors when creating a money request fails early on the backend. Manual requests can't really fail on the backend, but distance requests created while offline certainly can. This PR fixes the error handling in anticipation of that and is easier to test.

The most important change is to add some additional failure data when opening a failed money request report. Since that report doesn't exist we remove it and all data related to it. We also display an error message, and the same clean up runs when that message is dismissed.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/123`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/123#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/22126
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
**Set up**
Make all manual money requests fail by throwing at the [top of this function](https://github.com/Expensify/Web-Expensify/blob/429c9dddc457fb44d8fa365a6acaf443e27132a6/lib/TransactionAPI.php#L1044-L1045). Add this line.
```
throw new ExpError('12124', 'Test error');
```

**Test request and expense report creation failure t1**
1. Create a new workspace
2. Green plus > Request money
3. Select the manual tab, enter an amount, next
4. Select the newly created workspace
5. Click request
6. Verify that there is a red dot on the report preview and an error message is shown below
7. Verify that there is a red dot shown on the row for the expense report in the chat list
8. Click on the report preview
9. Verify that you reach a not found page
10. Go back to the workspace chat
11. Verify that the workspace chat is empty other than the welcome message
14. Verify that the row for the expense report in the chat list was removed

**Test request and expense report creation failure t2**
1. Create a new workspace
2. Green plus > Request money
3. Select the manual tab, enter an amount, next
4. Select the newly created workspace
5. Click request
6. Verify that there is a red dot on the report preview and an error message is shown below
7. Verify that there is a red dot shown on the row for the expense report in the chat list
1. Click the X to dismiss the error message
13. Verify that the report preview was removed
14. Verify that the row for the expense report in the chat list was removed

**Test only request creation failure**
1. Comment out the line that throws an error in the backend
2. Green plus > Request money
3. Select the manual tab, enter an amount, next
4. Select the newly created workspace
5. Click request
6. Uncomment the line to throw an error in the backend
7. Run the tests above and verify that they pass with a separate, failed, report preview.

- [ ] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
1. Run the tests above, going offline before creating the failed request, then going online after.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
15. Upload an image via copy paste
16. Verify a modal appears displaying a preview of that image
--->
No QA since there's no easy way to make the backend error. Later we will QA failure cases for distance requests.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.


### Screenshots/Videos
I only tested on web because the changes are platform independent.
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/26260477/954cdfb2-b44a-471d-9d4b-f6e79091d632


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

neil-marcellini committed:
Add RBR indicator on report preview

neil-marcellini committed:
Fix disable pay button and when report has errors

neil-marcellini committed:
Display money request transaction error messages

neil-marcellini committed:
Add transactionErrors prop type and default

neil-marcellini committed:
WIP cleanUpFailedMoneyRequest

neil-marcellini committed:
Fix selector for transaction errors

neil-marcellini committed:
Get last message text from actions

neil-marcellini committed:
WIP Clean up failed request with updated total

neil-marcellini committed:
Merge branch 'main' into neil-request-errors

neil-marcellini committed:
Remove unused prop after merge

neil-marcellini committed:
Remove duplicate error message

neil-marcellini committed:
Clean up money request when clearing action errors

neil-marcellini committed:
Fix crash by getting preview from workspace chat

neil-marcellini committed:
Merge branch 'main' into neil-request-errors

neil-marcellini committed:
Reset the report total after failed money request

neil-marcellini committed:
Merge branch 'main' into neil-request-errors

neil-marcellini committed:
Fix settlement button after merge

neil-marcellini committed:
WIP debug deleting action

neil-marcellini committed:
Fix deleting iou action, remove debugging

neil-marcellini committed:
Show error message when iouReport creation fails

thienlnam committed:
Merge branch 'main' into neil-request-errors

thienlnam committed:
Now that the onyx PR is merged, we can remove this workaround

thienlnam committed:
Merge branch 'main' into neil-request-errors

thienlnam committed:
Bump the onyx version

thienlnam committed:
Update types / delete the reportAction

thienlnam committed:
prettier

neil-marcellini committed:
Merge branch 'main' into neil-request-errors

neil-marcellini committed:
Style

neil-marcellini committed:
Delete preview for failed expense report on open

neil-marcellini committed:
Use existing clearReportActionErrors function

neil-marcellini committed:
Fix report preview error indicator

neil-marcellini committed:
Simplify money request clean up

neil-marcellini committed:
More clean up

neil-marcellini committed:
Merge branch 'main' into neil-request-errors

neil-marcellini committed:
Set report preview error the same as in chat list

neil-marcellini committed:
Undo useless changes to settlement button

neil-marcellini committed:
remove redundant boolean prop

neil-marcellini committed:
Fix check for creation error with null values

neil-marcellini committed:
Fix console error

neil-marcellini committed:
Ensure chat report's iouReport fields is reset

neil-marcellini committed:
Add JSDoc

neil-marcellini committed:
Fix lint again

neil-marcellini commented:
I'm going to put this up for review now, and then come back tomorrow to fix the unit tests.

neil-marcellini committed:
Linked report must exist before deleting actions

neil-marcellini committed:
Merge branch 'main' into neil-request-errors

neil-marcellini committed:
Fix missing code after merge

neil-marcellini commented:
@allroundexperts I think it will be pretty hard for you to test without being able to make a change on the backend so I'm going to remove you as a reviewer.

tgolen commented on src/types/onyx/ReportAction.ts:
```diff
@@ -52,6 +52,9 @@ type ReportActionBase = {
     /** The ID of the previous reportAction on the report. It is a string represenation of a 64-bit integer (or null for CREATED actions). */
     previousReportActionID?: string;
 
+    /** ID of a report */
```
Can this comment be more specific like 
```suggestion
    /** The ID of the report which the action belongs to */
```

neil-marcellini replied:
Updated

tgolen commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

tgolen commented:
> I only tested on web because the changes are platform independent.

Can you please complete the testing on all platforms? You've checked off all the checkboxes for testing them and there are no notes or explanations about why the tests were not performed.

Here were some bugs I found while testing.

`Bug 1:` No RBR in the LHN for the workspace chat after creating request.

![image](https://github.com/Expensify/App/assets/1228807/df6592a7-9883-48b3-8cd1-a3d57d01d3a7)



`Bug 2:` Create 2 failed requests, then 1 request that didn't fail. The expense report has two failed requests at the top, but you can only dismiss the error on one of them and the other one stays stuck. This anytime the first request is a failed one and the error on the first one can never be dismissed.

![image](https://github.com/Expensify/App/assets/1228807/021e4f3c-97de-46f2-a2f7-74e2669268d4)

![image](https://github.com/Expensify/App/assets/1228807/47939bfe-32b7-414c-ab60-1f8ef75aecf8)

`Bug 3:` In the state above (where there is 1 failed and 1 good request on a report), clicking on the failed request doesn't show a NOT FOUND page and you can view the details of it.

`Bug 4:` Similar to Bug 3, on a slow connection, the first time you access the details of the failed request, you can see the details. If you go back, then select the request again, there is a skeleton loading UI and then the NOT FOUND page appears.

neil-marcellini commented on src/libs/actions/Report.js:
```diff
@@ -516,6 +516,26 @@ function openReport(reportID, participantLoginList = [], newReportObject = {}, p
     ];
 
+    // If we're going to open a failed money request report, remove the report preview action in the failure data
+    const currentReport = ReportUtils.getReport(reportID);
```
Ok

neil-marcellini commented on src/libs/actions/Report.js:
```diff
@@ -516,6 +516,26 @@ function openReport(reportID, participantLoginList = [], newReportObject = {}, p
 
+    // If we're going to open a failed money request report, remove the report preview action in the failure data
+    const currentReport = ReportUtils.getReport(reportID);
+    const reportCreationErrors = _.chain(lodashGet(currentReport, 'errorFields.createChat')).compact().value();
```
Oh yeah good point. I think I chained because I thought that was needed to convert the object into a list. But I don't think chain does that anyways and compact can act on an object.

neil-marcellini committed:
Implement review feedback

neil-marcellini committed:
More descriptive field comment

neil-marcellini commented:
> > I only tested on web because the changes are platform independent.
> 
> Can you please complete the testing on all platforms? You've checked off all the checkboxes for testing them and there are no notes or explanations about why the tests were not performed.

That is the explanation for why I only tested on web. I guess it would a good idea to test on all platforms because some of this might behave differently on native due to styling, and because it's not a super simple flow.

neil-marcellini commented:
Thanks for the thorough review @tgolen! I addressed some of your feedback but I haven't started on the bugs.

I think this PR is in a bit of a weird place and I would like to chat about the overall plan more in Slack another day. For example I've seen it suggested that the RBR should direct to the lowest level context item (the individual request), vs the top level (workspace chat) as I have implemented now. Also, I would think that a failed money request should be shown on an existing expense report when there is one, but that's not the case. There are many more considerations and I would like to come up with a better plan to fix all of it.

tgolen commented:
I agree with you that this would be great to have more discussion about it. I think that the intentions were good for the current requirements, but when you actually try them out and see the UX, it becomes very clunky and difficult to navigate and recover from. I'll be on the lookout for that discussion to help out in whatever way I can!

tgolen commented:
@neil-marcellini what do you think about either closing this, putting it into draft state, or putting HOLD in the title?

tgolen commented:
@neil-marcellini what do you think about my previous suggestion?

> think about either closing this, putting it into draft state, or putting HOLD in the title?

neil-marcellini commented:
Oh yeah good point. I put it on HOLD (for planning) and marked it as a draft. After I get a chance to do a little pre-design I will either pick it back up or close it, depending on how different the chosen solution is.

neil-marcellini commented:
Super old and I'm not coming back to this lol 😢
